src/business_logic/transaction/objects/internal_deploy_account.rs

```diff
@@ -114,7 +118,7 @@ impl InternalDeployAccount {
         general_config: &StarknetGeneralConfig,
     ) -> Result<TransactionExecutionInfo, TransactionError>
     where
-        S: State + StateReader,
+        S: Clone + Default + State + StateReader,
```

Comment:
And please don't add `Clone` and `Default` either. Those were also removed for a reason: they pretty much make implementing _real_ state implementations very hard.
Please see https://github.com/lambdaclass/starknet_in_rust/pull/271 and https://github.com/lambdaclass/starknet_in_rust/pull/367 for details.

Reply:
Yes, I agree cc @SantiagoPittella @mmsc2 @fmoletta

Reply:
Nice, I just pushed a new commit removing those, and replacing the references in the return values for owned values.